core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/contribution/ActivateDiagramFiltersContribution.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2022 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.data.migration.contribution;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.sirius.diagram.DDiagram;
+import org.eclipse.sirius.diagram.description.filter.FilterDescription;
+import org.polarsys.capella.common.ef.ExecutionManager;
+import org.polarsys.capella.core.data.migration.context.MigrationContext;
+import org.polarsys.capella.core.model.handler.command.CapellaResourceHelper;
+import org.polarsys.capella.core.sirius.analysis.constants.IFilterNameConstants;
+import org.polarsys.capella.core.sirius.analysis.helpers.DDiagramHelper;
+
+/**
+ * This class activate filters for computed transitions on MSM diagrams.
+ */
+public class ActivateDiagramFiltersContribution extends AbstractMigrationContribution {
+
+  @Override
+  public void unaryEndMigrationExecute(ExecutionManager executionManager, Resource resource, MigrationContext context) {
+    super.unaryEndMigrationExecute(executionManager, resource, context);
+    activateRelevantDiagramFilters(resource);
+  }
+
+  private void activateRelevantDiagramFilters(Resource resource) {
+    for (DDiagram diagram : getAllMSMDiagrams(resource)) {
```

Comment:
can you enable activation of filter only if model is not in 6 version ?
(if major < 6)

(something like https://github.com/eclipse/capella/blob/409a90c4a2ed6c9c355badee0d068c32ec47f195/core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/contribution/EnableFCAndPPFiltersMigrationContribution.java#L47-L62 )